lib/CType.ml

```diff
@@ -100,37 +103,45 @@ let rec is_pte t = match t with
 
 let rec is_array = function
   | Array _ -> true
-  | Atomic t|Volatile t -> is_array t
-  | _ -> false
+  | Atomic t | Volatile t | Const t -> is_array t
+  | Base _ | Pointer _ -> false
 
 let rec is_atomic = function
-  | Volatile t -> is_atomic t
+  | Volatile t | Const t -> is_atomic t
   | Atomic _ -> true
-  | _ -> false
-
-let strip_atomic0 = function
-  | Atomic t -> t
-  | t -> t
+  | Array _ | Base _ | Pointer _ -> false
 
 let rec strip_atomic = function
   | Volatile t -> Volatile (strip_atomic t)
-  | t ->  strip_atomic0 t
-
-let strip_volatile0 = function
-  | Volatile t -> t
-  | t -> t
+  | Atomic t -> strip_atomic t (* We handle the case where we have malformed types *)
+  | Const t -> Const (strip_atomic t)
+  | Pointer t -> Pointer (strip_atomic t)
```

Comment:
again this changes the behavior of `strip_volatile` I see why we may want to do it but I am not sure it will ever be used.

Reply:
do you feel strongly about this? I am happy to remove it, but this way we get a guarantee that volatile is indeed removed in the pathological case. since CType is an open library I think we should assume the user might accidentally give the function a poorly formed type

Reply:
No let's take this conversation upstream.